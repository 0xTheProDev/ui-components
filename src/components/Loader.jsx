// @flow
import React from 'react';
import cn from 'classnames';

export const StandardLoader = ({
  large,
  small,
  dark,
}: { large?: boolean, small?: boolean, dark?: boolean }) => (
  <svg>
    <svg
      className={cn('sg-loader', {
        'is-small': small,
        'is-large': large,
        'loader-on-dark': dark,
      })}
    >
      <rect className="logo-square bottom-left" />
      <rect className="logo-square bottom-middle" />
      <rect className="logo-square middle" />
      <rect className="logo-square top-middle" />
      <rect className="logo-square top-right" />
      <rect className="logo-square middle-left" />
      <rect className="logo-square middle-right" />
    </svg>
  </svg>
);

StandardLoader.defaultProps = {
  large: false,
  small: false,
  dark: false,
};

export const FragmentLoader = ({
  large,
  small,
  dark,
}: { large?: boolean, small?: boolean, dark?: boolean }) => (
  <svg
    className={cn('fragment-loader', {
      'is-small': small,
      'is-large': large,
      'loader-on-dark': dark,
    })}
  >
    <rect className="fragment-square left" />
    <rect className="fragment-square middle" />
    <rect className="fragment-square right" />
  </svg>
);

FragmentLoader.defaultProps = {
  large: false,
  small: false,
  dark: false,
};
