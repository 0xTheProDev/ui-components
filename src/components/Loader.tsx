import * as React from 'react';
import * as cn from 'classnames';

export type LoaderProps = {
  onDark?: boolean;
  large?: boolean;
  small?: boolean;
  centered?: boolean;
};

const Loader: React.SFC<LoaderProps> = ({ onDark, large, small, centered }) => {
  let size = 10;
  if (large) size *= 2;
  if (small) size /= 2;
  return (
    <svg
      className={cn('sg-loader', {
        'loader-on-dark': onDark,
        'is-large': large,
        'is-small': small,
        'is-centered': centered,
      })}
    >
      <rect className="logo-square bottom-left" width={size} height={size} />
      <rect className="logo-square bottom-middle" width={size} height={size} />
      <rect className="logo-square middle" width={size} height={size} />
      <rect className="logo-square top-middle" width={size} height={size} />
      <rect className="logo-square top-right" width={size} height={size} />
      <rect className="logo-square middle-left" width={size} height={size} />
      <rect className="logo-square middle-right" width={size} height={size} />
    </svg>
  );
};

Loader.defaultProps = {
  onDark: false,
  large: false,
  small: false,
  centered: false,
};

export default Loader;
