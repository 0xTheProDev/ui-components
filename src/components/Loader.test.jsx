// @flow
import React from 'react';
import { shallow } from 'enzyme';
import { StandardLoader, FragmentLoader } from './Loader';

describe('StandardLoader', () => {
  it('should work with a "small" attribute', () => {
    const wrapper = shallow(
      <StandardLoader small />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should work with a "large" attribute', () => {
    const wrapper = shallow(
      <StandardLoader large />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should work with a "small" and "dark" attributes', () => {
    const wrapper = shallow(
      <StandardLoader small dark />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});

describe('FragmentLoader', () => {
  it('should work with a "small" attribute', () => {
    const wrapper = shallow(
      <FragmentLoader small />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should work with a "large" attribute', () => {
    const wrapper = shallow(
      <FragmentLoader large />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should work with a "small" and "dark" attributes', () => {
    const wrapper = shallow(
      <FragmentLoader small dark />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
