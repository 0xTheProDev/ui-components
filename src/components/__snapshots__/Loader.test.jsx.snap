// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FragmentLoader should work with a "large" attribute 1`] = `
<svg
  className="fragment-loader"
>
  <rect
    className="fragment-square left"
  />
  <rect
    className="fragment-square middle"
  />
  <rect
    className="fragment-square right"
  />
</svg>
`;

exports[`FragmentLoader should work with a "small" and "dark" attributes 1`] = `
<svg
  className="fragment-loader loader-on-dark"
>
  <rect
    className="fragment-square left"
  />
  <rect
    className="fragment-square middle"
  />
  <rect
    className="fragment-square right"
  />
</svg>
`;

exports[`FragmentLoader should work with a "small" attribute 1`] = `
<svg
  className="fragment-loader"
>
  <rect
    className="fragment-square left"
  />
  <rect
    className="fragment-square middle"
  />
  <rect
    className="fragment-square right"
  />
</svg>
`;

exports[`StandardLoader should work with a "large" attribute 1`] = `
<svg
  className="sg-loader"
>
  <rect
    className="logo-square bottom-left"
  />
  <rect
    className="logo-square bottom-middle"
  />
  <rect
    className="logo-square middle"
  />
  <rect
    className="logo-square top-middle"
  />
  <rect
    className="logo-square top-right"
  />
  <rect
    className="logo-square middle-left"
  />
  <rect
    className="logo-square middle-right"
  />
</svg>
`;

exports[`StandardLoader should work with a "small" and "dark" attributes 1`] = `
<svg
  className="sg-loader loader-on-dark"
>
  <rect
    className="logo-square bottom-left"
  />
  <rect
    className="logo-square bottom-middle"
  />
  <rect
    className="logo-square middle"
  />
  <rect
    className="logo-square top-middle"
  />
  <rect
    className="logo-square top-right"
  />
  <rect
    className="logo-square middle-left"
  />
  <rect
    className="logo-square middle-right"
  />
</svg>
`;

exports[`StandardLoader should work with a "small" attribute 1`] = `
<svg
  className="sg-loader"
>
  <rect
    className="logo-square bottom-left"
  />
  <rect
    className="logo-square bottom-middle"
  />
  <rect
    className="logo-square middle"
  />
  <rect
    className="logo-square top-middle"
  />
  <rect
    className="logo-square top-right"
  />
  <rect
    className="logo-square middle-left"
  />
  <rect
    className="logo-square middle-right"
  />
</svg>
`;
